### PR TITLE
Add pg_clients test for serverless driver

### LINKED ISSUE #3819: Test `@neondatabase/serverless` driver in pg_clients
We have a test suite for testing different postgres clients against cloud Neon (https://github.com/neondatabase/neon/tree/main/test_runner/pg_clients).

Our serverless driver (https://github.com/neondatabase/serverless) supports running in NodeJS, with just `npm install @neondatabase/serverless && node index.js`.

We should add a test to verify that this setup is working and there is no issues in our proxy and serverless driver. We should run this test on staging, to catch bugs earlier than they are deployed to production.

### PR DESCRIPTION
**Pull request by petuhovskiy**
## Issue ticket number and link

Fixes #3819 

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

### CODE REVIEW

**Review by bayandin — APPROVED** [review]
Nice!

Tests passed https://github.com/neondatabase/neon/actions/runs/4426419936/jobs/7762685620#step:8:190

File: test_runner/pg_clients/typescript/serverless-driver/package.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "module",
+  "dependencies": {
+    "@neondatabase/serverless": "^0.2.8",
+    "ws": "^8.13.0"
```
[the remark this answers is not in the document]
Comment (petuhovskiy) [reply test_runner/pg_clients/typescript/serverless-driver/package.json:5]: We do, `@neondatabase/serverless` uses it internally to work from node.js

[the remark this answers is not in the document]
Comment (bayandin) [reply test_runner/pg_clients/typescript/serverless-driver/package.json:5]: Shouldn't it be a direct dependency of  `@neondatabase/serverless` then?

[the remark this answers is not in the document]
Comment (bayandin) [reply test_runner/pg_clients/typescript/serverless-driver/package.json:5]: Got it, thanks for the explanation!

**Pull request merged**